### PR TITLE
Add rice

### PR DESCRIPTION
**Pull request by ornobalam**
Added the species Oryza sativa, along with one demographic model from a 2007 paper.

### CODE REVIEW

**Comment by nspope** [comment]
Fantastic, thanks @ornobalam! We'll find a volunteer to QC these.

**Comment by ornobalam** [comment]
Awesome, thank you!

**Comment by gregorgorjanc** [comment]
I can ask Ines to look at this! Is there a guidance doc I could share with her? If she has time!

**Comment by nspope** [comment]
Thanks @gregorgorjanc, that'd be awesome. I'll merge this in once I check formatting, etc, then open a QC issue. The rest of the process is outlined in [this](https://popsim-consortium.github.io/stdpopsim-docs/latest/development.html#overview-of-the-stdpopsim-review-process) section of the developer docs.  @igronau may also be able to provide some guidance.

**Comment by igronau** [comment]
@gregorgorjanc - Ines should follow the steps in the developers' doc (mentioned in @nspope 's comment). The QC issue will have a checklist of things to guide the process. For every parameter specified in the ``species.py`` file you need to to the following:

1. Find the paper used as reference for that parameter (should be specified using a citation object)
2. Use the documentation in the ``species.py`` file to figure out what the value of the parameter should be. Avoid looking at the values actually specified in the ``species.py`` file. If something is not clear in this process, you should tag the original contributor in the QC issue to clarify.
3. Write in the value(s) in the code of the test file for the species.

The important thing is to be **very critical** and don't be shy in calling into question any of the decisions made in the implementation. This could be the last time someone takes a critical look at the implementation.

I can help Ines through this process if needed.

**Comment by petrelharp** [comment]
> The important thing is to be very critical

... in a nice way. We mean "be very detail-oriented".

**Comment by igronau** [comment]
point taken :-).

File: stdpopsim/catalog/OrySat/demographic_models.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
+import msprime
+import stdpopsim
+
+_species = stdpopsim.get_species("OrySat")
+
+# population definitions that are reused.
+_rufipogon = stdpopsim.Population(id="RUF", description="Oryza rufipogon (wild rice)")
+_indica = stdpopsim.Population(id="IND", description="Oryza sativa indica")
+_tropical_japonica = stdpopsim.Population(
+    id="TRP", description="Tropical Oryza sativa japonica"
```
Comment (minesrebollo) [inline stdpopsim/catalog/OrySat/demographic_models.py:10]: I would use "TRJ" for the tropical japonica id.

**Comment by minesrebollo** [comment]
Sure, I will look at it, hopefully by the end of this week/early next week :)

File: stdpopsim/catalog/OrySat/demographic_models.py (hunk 1 above)
Comment (ornobalam) [reply stdpopsim/catalog/OrySat/demographic_models.py:10]: Okay

**Comment by nspope** [comment]
Looks like the species test stubs didn't make it into these commits, @ornobalam -- that is, running ``python -m maintenance add-species oryza_sativa`` from the base of the repository should have also generated the file ``tests/test_OrySat.py`` that contains some stubs that are filled out during QC. Would mind adding this file to the PR, please?

If you need to rerun the maintenance utility and it complains about OrySat already existing, then you'll need to make a new (temporary) branch; re-run the ``python -m maintenance add-species oryza_sativa`` bit; and copy the newly-generated ``tests/test_OrySat.py`` into your feature branch (add_rice). Sorry for the hassle!

Then I can merge this & make a QC issue that @minesrebollo will add comments to, regarding parameter values, implementation, etc.

**Comment by nspope** [comment]
Oh, there's one more thing @ornobalam -- could you please add `docs/parameter_tables/OrySat/BottleneckMigration_3C07.csv` with the demographic model parameters, as described [here](https://popsim-consortium.github.io/stdpopsim-docs/latest/development.html#adding-a-parameter-table-to-the-docs)? Thanks!

**Comment by ornobalam** [comment]
Hi, thanks for the instructions! Will get these done by Monday!

Ornob

File: tests/test_OrySat.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+
+class TestSpeciesData(test_species.SpeciesTestBase):
+    species = stdpopsim.get_species("OrySat")
+
+    def test_ensembl_id(self):
+        assert self.species.ensembl_id == "oryza_sativa"
+
+    def test_name(self):
+        assert self.species.name == "Oryza sativa Japonica Group"
```
Comment (nspope) [inline tests/test_OrySat.py:14]:
```suggestion
        assert self.species.name == "Oryza sativa"
```

**Comment by ornobalam** [comment]
Thanks! I've made them consistent now, I think.

**Comment by nspope** [comment]
Thanks @ornobalam!

@minesrebollo I'll open separate QC issues for the species and demographic model. Please comment on those with any issues that you see with the implemented parameters; and feel free to reach out to @igronau or myself with questions. Thanks much!

**Pull request merged**